### PR TITLE
restart a heroku app based on a webhook

### PR DESCRIPTION
**Pull request by hiattp**

### CODE REVIEW

**Comment by janklimo** [comment]
@hiattp I like this 👍 can you give me access to the app on Heroku? I have the search and webhook to it ready. See [here](https://papertrailapp.com/alerts/1457321/edit) if you want to update it with ID.

**Comment by hiattp** [comment]
Oops yeah thought you would have access automatically but you do now. The app is deployed in the PG account so you should already have access there (I think?). Let me know if not

**Comment by janklimo** [comment]
got it, thanks

File: app/models/heroku_application.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class HerokuApplication < ApplicationRecord
```
Comment (NevilleS) [inline app/models/heroku_application.rb:1]: nit: It'd probably make sense to just add some defensive validations to ensure that any application we create has a name, oauth token, grace period defined.

Also would be kinda good to be able to test that the key works (ie. can authenticate) before it tries to actually trigger a restart.

Not blockers to getting this configured though and running it.

**Review by NevilleS — COMMENTED** [review]
I like it, @hiattp, let's get it running

File: app/models/heroku_application.rb (hunk 1 above)
Comment (hiattp) [reply app/models/heroku_application.rb:1]: 👍 added those to https://github.com/lockstep/heroku_memory_capitulator/issues/2

**Pull request merged**